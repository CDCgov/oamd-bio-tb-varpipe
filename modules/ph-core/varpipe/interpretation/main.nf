process VARPIPE_INTERPRETATION {

    tag "${meta.id}"
    label 'process_single'

    container 'oamd-bio-biocontainers-python:3.12_80c558c_v3'

    input:
    tuple val(meta), path(summary), path(structuralVariants), path(drLociAnnotation), path(targetRegionCoverage)
    path(reported)

    output:
    tuple val(meta), path("*_summary.txt"), emit: summary
    tuple val(meta), path("*_interpretation.txt"), emit: interpretation
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    template 'interpret.py'

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_summary.txt
    touch ${prefix}_interpretation.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version | cut -d' ' -f2)
    END_VERSIONS
    """
}
