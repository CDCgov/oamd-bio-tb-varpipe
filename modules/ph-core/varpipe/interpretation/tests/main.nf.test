nextflow_process {

    name "Test Process VARPIPE_INTERPRETATION"
    script "../main.nf"
    process "VARPIPE_INTERPRETATION"

    tag "modules"
    tag "modules_phcore"
    tag "varpipe"
    tag "varpipe/interpretation"

    test("varpipe interpret report [stats] [coverage] [final annotation] [variants]") {

        when {
            process {
                """
                // Prepare input data
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/coverage/summary.txt'),
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/coverage/test_structural_variants.txt'),
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/annotation/test_DR_loci_annotation.txt'),
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/coverage/test_target_region_coverage.txt')
                ]
                input[1] = file(params.modules_testdata_base_path + '/genomics/tuberculosis/genome/kb/varpipe/reportable.txt')
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("varpipe interpret report [stats] [coverage] [final annotation] [variants] -stub") {

        options "-stub"

        when {
            process {
                """
                // Prepare input data
                input[0] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/coverage/summary.txt'),
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/coverage/test_structural_variants.txt'),
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/annotation/test_DR_loci_annotation.txt'),
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/coverage/test_target_region_coverage.txt')
                ]
                input[1] = file(params.modules_testdata_base_path + '/genomics/tuberculosis/genome/kb/varpipe/reportable.txt')
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
