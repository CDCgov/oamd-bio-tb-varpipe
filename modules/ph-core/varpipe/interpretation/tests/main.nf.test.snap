{
    "varpipe interpret report [stats] [coverage] [final annotation] [variants] -stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_summary.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test_interpretation.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,5f5124f6054e5ef8b5362c3aa3942912"
                ],
                "interpretation": [
                    [
                        {
                            "id": "test"
                        },
                        "test_interpretation.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "summary": [
                    [
                        {
                            "id": "test"
                        },
                        "test_summary.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,5f5124f6054e5ef8b5362c3aa3942912"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-25T16:45:54.557357122"
    },
    "varpipe interpret report [stats] [coverage] [final annotation] [variants]": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_summary.txt:md5,339139edb768113c69eb3b44fcb33e7a"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test_interpretation.txt:md5,498224fc10466af49f4119ebe44e8824"
                    ]
                ],
                "2": [
                    "versions.yml:md5,8330a9d66b2955ab2512684d4b2f1628"
                ],
                "interpretation": [
                    [
                        {
                            "id": "test"
                        },
                        "test_interpretation.txt:md5,498224fc10466af49f4119ebe44e8824"
                    ]
                ],
                "summary": [
                    [
                        {
                            "id": "test"
                        },
                        "test_summary.txt:md5,339139edb768113c69eb3b44fcb33e7a"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8330a9d66b2955ab2512684d4b2f1628"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-05-14T11:15:10.344659867"
    }
}
